tools/add_triage_signature.py

```diff
@@ -48,33 +53,60 @@
 """
 
 
+def config_logger(verbose: bool):
+    global logger
+
+    handler = colorlog.StreamHandler()
+    handler.setFormatter(colorlog.ColoredFormatter('%(log_color)s%(levelname)-10s %(message)s'))
+
+    logger.setLevel(logging.INFO)
+    logger.addHandler(handler)
+
+    if verbose:
+        logger.setLevel(logging.DEBUG)
+
+
 def format_description(failure_data):
     return JIRA_DESCRIPTION.format(**failure_data)
 
+
 def days_ago(datestr):
     try:
         return (datetime.now() - dateutil.parser.isoparse(datestr)).days
     except Exception:
         logger.debug("Cannot parse date: %s", datestr)
         return 9999
 
+
 @functools.lru_cache(maxsize=1000)
 def get_metadata_json(cluster_url):
     res = requests.get("{}/metdata.json".format(cluster_url))
     res.raise_for_status()
     return res.json()
 
+
 @functools.lru_cache(maxsize=1000)
 def get_events_json(cluster_url, cluster_id):
     res = requests.get(f"{cluster_url}/cluster_{cluster_id}_events.json")
     res.raise_for_status()
     return res.json()
 
+
+@functools.lru_cache(maxsize=1000)
```

Comment:
no point in caching this function if you're not returning any values.

If anything, it should stop accepting an `output_file` and just cache the results so they can be reused by all signatures that need logs

Reply:
It should probably already have support for extracting from host tars as well.

Basically get rid of the entire:
```
        with NamedTemporaryFile() as temp:
            try:
                get_cluster_logs(url, cluster_id, temp.name)
            except Exception as e:
                logger.error("Error getting logs for %s at %s: %s, they may have been deleted", issue_key, url, e)
                return

            with TemporaryDirectory() as tempdir:
                with tarfile.open(temp.name) as tar:
                    tar.extractall(tempdir)

                hosts = []

                for tar_file in os.listdir(tempdir):
                    tar_file = os.path.join(tempdir, tar_file)
                    try:
                        dmesg_file = extract_file_from_tar(tar_file, "dmesg", tempdir)
                    except FileNotFoundError:
                        continue

                    host_id = get_host_id_for_host_tar(tar_file)
```


boilerplate each signature has to do just to extract logs from hosts

Reply:
Seems to be a lot of work.
Opened a new ticket just for that - https://issues.redhat.com/browse/MGMT-4681